portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,33 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the value entered in the form.
+    String message = request.getParameter("message");
+    String number = request.getParameter("number");
+    String email = request.getParameter("email");
+    String name = request.getParameter("name");
+
+
+    // Print the value so you can see it in the server logs.
+    System.out.println("------------------------");
+    System.out.println("Name: " + name);
+    System.out.println("Email: " + email);
+    System.out.println("Number " + number);
+    System.out.println("\nMessage: " + message);
+    System.out.println("------------------------");
+
+    // redirects back to top of website
+    response.sendRedirect("https://drodrigues-sps-spring21.uc.r.appspot.com");
```

Comment:
Bonus task if (and only if) you're interested - since this is a react webapp, a big advantage is not having to reload the website just to update the content. If you wanted to, you could make the response send back a bit of json or something indicating whether the stuff was logged successfully, and then have your client-side code just print a "submitted!" message or similar. Just a thought, this is of course only if you want to give yourself an extra challenge ;)

Reply:
That sounds great! I will scratch off the client check and try to do it on the server. I am assuming this would include fetching to get the JSON data from the servlet? 

As for the reload, if I am not mistaken, the `FormHandlerServlet.java` servlet takes the user to another page even if a redirect is not called, correct? If so, then I will have to find a way to prevent this behavior or change it in a way it does not refresh the page.

This just got a bit more interesting and a challenge I would gladly take 🙂

Reply:
Sweet! (Of course you should still prioritize keeping up with the current assignments - make sure you get those done first, this is just an optional bonus challenge!)

> I am assuming this would include fetching to get the JSON data from the servlet?
It depends what you mean here. You don't actually need to make a separate request/response handler for this. I'll explain in a moment, but the answer relates to my response to your next question:

> the FormHandlerServlet.java servlet takes the user to another page even if a redirect is not called, correct?
Not exactly, unless there's some Java-servlet-specific thing going on that I'm not aware of. The HttpServletResponse returned from this function is just like the one returned from your RandomQuoteServlet you did earlier. (I would expect that if you called response.sendRedirect in your randomQuoteServlet it would take the user to another page in the same way that this one currently does.) If you write json data to your response here, you should be able to read that data on the frontend in the same way you do for the random quotes!

In short: doGet and doPost handlers behave in pretty much exactly the same way, the only difference (by convention) is that POST means something should change on the server, and GET means that you're only retrieving data of some kind. If you're interested in learning more about this, ask me sometime about the HTTP protocol in general, other HTTP methods (PUT, for example), and REST vs SOAP.

Reply:
Thank you so much for the in-depth explanation, that sounds a lot easier than what I had in mind! I will definitely be asking those questions in our Friday meeting since that is a subject that I am not at all experienced in. I will remove the redirect and see if the behavior persists and try to send back JSON to trigger a success or failure alert after the data check is completed 🙂  I will make that branch once I am finished with this week's project!